lib/src/declarations.dart

```diff
@@ -43,24 +31,27 @@ abstract class MethodDeclaration implements MethodType {
   Iterable<TypeParameterDeclaration> get typeParameters;
 }
 
-abstract class TargetMethodDeclaration implements MethodDeclaration {
-  void addToClass(Code declaration);
-}
-
 abstract class FieldDeclaration implements FieldType {
   String get name;
 
   TypeDeclaration get type;
 }
 
-abstract class TargetFieldDeclaration implements FieldDeclaration {
-  void addToClass(Code declaration);
-}
-
 abstract class ParameterDeclaration implements ParameterType {
   TypeDeclaration get type;
 }
 
 abstract class TypeParameterDeclaration implements TypeParameterType {
   TypeDeclaration? get bounds;
 }
+
+abstract class DeclarationBuilder {
```

Comment:
Nice ya I like the `Builder` name here a lot, having the separate interface is nice just for a better name haha.